application/common/models/User.php

```diff
@@ -20,6 +20,9 @@
 use yii\db\conditions\OrCondition;
 use yii\helpers\ArrayHelper;
 
+/**
+ * @property EmailLog[] $emailLogs
```

Comment:
I _think_ the only place the `EmailLog` properity would be used is in the `beforeDelete` method. It may be simpler, for both implementation and maintenance, to just update that method with the appropriate query.

Reply:
I'd like to keep using Yii's built-in capability for handling those model-to-model relations. I originally defined this as a foreign key, but realized that wouldn't work well (for various reasons), but that's where I got the code for how to set up the code so that `User` models can easily get the list of `emailLogs` that match that User's (current) email address.